driver/bpf/filler_helpers.h

```diff
@@ -212,10 +215,10 @@ static __always_inline bool bpf_getsockname(struct socket *sock,
 	return true;
 }
 
-static __always_inline int bpf_addr_to_kernel(void *uaddr, int ulen,
+static __always_inline int bpf_addr_to_kernel(void *uaddr, unsigned long ulen,
 					      struct sockaddr *kaddr)
 {
-	if (ulen < 0 || ulen > sizeof(struct sockaddr_storage))
+	if (ulen > BPF_MAX_VAR_SIZE)
```

Comment:
This seems to increase the valid range of ulen. Previously the function would return `-EINVAL` if `ulen > sizeof(struct sockaddr_storage)`, but on my machine that `sizeof` operation resolves to 128, which is far, far less than `BPF_MAX_VAR_SIZE`. Is this intended?

Reply:
Looks like we're doing `ulen & 0xff` (more or less) below. I feel that we should return an error whenever `ulen > 0xff` (otherwise the caller can request e.g. 0x1ff bytes--the value comes straight from userspace--while we copy 0xff and return success).

The original check against `sizeof(struct sockaddr_storage)` feels more correct.

Also, not sure what exactly BPF_FORBIDS_ZERO_ACCESS implies but if my guess (length must be > 0) is correct, both the `#ifdef` and the `& 0xff` (if we revert the size check) is unnecessary (ulen will always be > 0 after line 224 and < 0xff after original line 218).

Though I expect both are there to keep the verifier happy?

Reply:
That check seems like a leftover from [move_addr_to_kernel](https://elixir.bootlin.com/linux/latest/source/net/socket.c#L192) kernel function.

Anyways, we read at maximum `ulen & 0xff` every time (see `bpf_probe_read` below).

After some intensive and day-long testing, we (@fntlnz and I) have not been able to break it.

Reply:
@gnosek The verifier rejects a check against `sizeof(struct sockaddr_storage)`.

Reply:
Anyway the bug here was about the `int` (now `unsigned long`) parameter and the missing check.

Both were required by the verifier (giving two different errors on each of them). Thus, this is the scope of this PR.

In case we'd like to optimize the behaviour of the mask and the presence (or not) of ifdefs, we should approach those in another PR, imho.

As per our testing, they do not interfere with the correct functioning of the probe.

Reply:
So if you replace `BPF_MAX_VAR_SIZE` with `sizeof(struct sockaddr_storage)`, the verifier fails?

That is bizarre. That seems like a bug in the verifier.

Reply:
Yes, we tried in many many ways to retain that check too, but the verifier rejects it.